lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -154,6 +154,16 @@ def unsafe_make_column_not_nullable(table, column, options={}) # options arg is
     end
   end
 
+  def safe_add_index_on_empty_table(table, columns, options={})
+    table_struct = PgHaMigrations::Table.from_table_name(table)
+
+    if connection.select_value("SELECT EXISTS (SELECT 1 FROM #{table_struct.fully_qualified_name} LIMIT 1)")
+      raise PgHaMigrations::InvalidMigrationError, "Table #{table_struct.inspect} has rows. Please use safe_add_concurrent_index instead."
+    end
```

Comment:
So I don't think this is 100% guaranteed to be safe, as the application _could_ write something to the table in the split second after this query runs. But in practice, I think that would be pretty rare

Reply:
Also, it would have to insert a _ton_ of data to matter.

Reply:
Actually, what if we re-checked this under the lock? Then it would be guaranteed to be safe.

Reply:
Oh yeah that's a good idea